src/coreclr/pgosupport.cmake

```diff
@@ -52,17 +51,17 @@ function(add_pgo TargetName)
                 add_compile_definitions(WITH_NATIVE_PGO)
             else(CLR_CMAKE_HOST_WIN32)
                 if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
-                    if((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16))
-                        if(HAVE_LTO)
+                    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+                        check_have_lto_and_pgodata_supported(${ProfilePath})
+                        if(HAVE_LTO_AND_PGO_DATA_SUPPORTED)
+                            message(STATUS "Enabling profile guided optimizations for ${TargetName}")
                             target_compile_options(${TargetName} PRIVATE -flto -fprofile-instr-use=${ProfilePath} -Wno-profile-instr-out-of-date -Wno-profile-instr-unprofiled)
                             set_property(TARGET ${TargetName} APPEND_STRING PROPERTY LINK_FLAGS " -flto -fprofile-instr-use=${ProfilePath}")
                             add_compile_definitions(WITH_NATIVE_PGO)
-                        else(HAVE_LTO)
-                            message(WARNING "LTO is not supported, skipping profile guided optimizations")
-                        endif(HAVE_LTO)
-                    else((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16))
-                        message(WARNING "PGO is not supported; Clang 16 or later is required for profile guided optimizations")
-                    endif((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16))
+                        else(HAVE_LTO_AND_PGO_DATA_SUPPORTED)
+                            message(WARNING "LTO is not supported or PGO optimization data not compatible, skipping profile guided optimizations for ${TargetName}")
```

Comment:
If we pass msbuild's `OfficialBuildId != ''` to cmake as `CLR_IS_OFFICIAL_BUILD` e.g., then we can use it to error out to make the problem visible (probably we don't want it to slip through unnoticed).

Reply:
good idea, but that is also set for source-build and afaik they can't use our PGO packages so we shouldn't error in that case. I'd do that in a separate PR.